systemtest/src/test/java/io/strimzi/systemtest/log/LogSettingST.java

```diff
@@ -329,6 +341,162 @@ private String configMap(String configMapName) {
         return configMapData.get(configMapKey);
     }
 
+    @Test
+    @Order(14)
+    @SuppressWarnings({"checkstyle:MethodLength"})
+    void testDynamicallySetEOloggingLevels() throws InterruptedException {
+        String eoDeploymentName = KafkaResources.entityOperatorDeploymentName(CLUSTER_NAME);
+        Map<String, String> eoPods = DeploymentUtils.depSnapshot(eoDeploymentName);
+
+        final String eoPodName = eoPods.keySet().iterator().next();
+
+        LOGGER.info("Setting log level of TO and UO to OFF - no records should appear in log");
+        // change inline logging
+        InlineLogging ilOff = new InlineLogging();
+        ilOff.setLoggers(Collections.singletonMap("rootLogger.level", OFF));
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, k -> {
+            k.getSpec().getEntityOperator().getTopicOperator().setLogging(ilOff);
+            k.getSpec().getEntityOperator().getUserOperator().setLogging(ilOff);
+        });
+
+        LOGGER.info("The EO shouldn't roll - verifying pod stability");
+        DeploymentUtils.waitForNoRollingUpdate(eoDeploymentName, eoPods);
+
+        LOGGER.info("Waiting for log4j2.properties will contain desired settings");
+        TestUtils.waitFor("Logger change", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
+            () -> cmdKubeClient().execInPodContainer(eoPodName, "topic-operator", "cat", "/opt/topic-operator/custom-config/log4j2.properties").out().contains("rootLogger.level=OFF")
+                && cmdKubeClient().execInPodContainer(eoPodName, "user-operator", "cat", "/opt/user-operator/custom-config/log4j2.properties").out().contains("rootLogger.level=OFF")
+        );
+
+        LOGGER.info("Waiting {} ms for INFO log will disappear", RECONCILIATION_INTERVAL * 2);
+        //wait some time if TO and UO will log something
+        Thread.sleep(RECONCILIATION_INTERVAL * 2);
+
+        LOGGER.info("Asserting if log is without records");
+        assertThat(StUtils.getLogFromPodByTime(eoPodName, "topic-operator", "30s"), is(emptyString()));
+        assertThat(StUtils.getLogFromPodByTime(eoPodName, "user-operator", "30s"), is(emptyString()));
+
+        LOGGER.info("Changing rootLogger level to INFO with inline logging");
+        InlineLogging ilInfo = new InlineLogging();
+        ilInfo.setLoggers(Collections.singletonMap("rootLogger.level", INFO));
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, k -> {
+            k.getSpec().getEntityOperator().getTopicOperator().setLogging(ilInfo);
+            k.getSpec().getEntityOperator().getUserOperator().setLogging(ilInfo);
+        });
+
+        LOGGER.info("The EO shouldn't roll - verifying pod stability");
+        DeploymentUtils.waitForNoRollingUpdate(eoDeploymentName, eoPods);
```

Comment:
Rather than adding these calls (which AFAICS just slow the test down) can't you take a pod snapshot (`DeploymentUtils.depSnapshot()`) at the start and verify that it's the same at the end?